### PR TITLE
[CS2] Fix #3098: Suppressed newline should be unsuppressed by semicolon

### LINKED ISSUE #3098: Lexer: `a = ; 5`: Suppressed newline should be unsuppressed by semicolon
(from #3053)

``` coffee
# rule : 
'Assignable = TERMINATOR Expression'
# supposedly allowing that :
a =
5
# but also allows this
a = ; 5
```

### PR DESCRIPTION
**Pull request by GeoffreyBooth**
Fixes #3098. Now an error is thrown if a semicolon follows an `=` or anything in the `UNFINISHED` group of tokens. No tests fail as a result of this PR, so presumably this isn’t a breaking change (aside from the case of `a = ; 5` which we want to throw an error).

```coffee
> ./bin/coffee -bpe 'a = ; 5'
[stdin]:1:5: error: unexpected ;
a = ; 5
    ^
```

### CODE REVIEW

File: src/lexer.coffee [hunk 1]
```diff
@@ -662,6 +662,7 @@ exports.Lexer = class Lexer
       @exportSpecifierList = no
 
     if value is ';'
+      @error 'unexpected ;' if prev?[0] in ['=', UNFINISHED...]
```
Comment (vendethiel) [inline src/lexer.coffee]: do we not want `=` in `UNFINISHED`?

Comment (vendethiel) [reply src/lexer.coffee]: or maybe the other way around: what does adding `in UNFINISHED` fixes, given that `prev?[0] is '='` is the relevant fix here? Same in `suppressSemicolons`.

Comment (GeoffreyBooth) [reply src/lexer.coffee]: See https://github.com/jashkenas/coffeescript/issues/3098#issuecomment-26680094

`UNFINISHED` covers all the similar tokens like `+=` etc. I think @jashkenas in his comment was worried about the special cases around `=`, like `=(newline)(object literal)`.

Comment (GeoffreyBooth) [reply src/lexer.coffee]: Also, putting `=` into `UNFINISHED` causes tests to fail.

Comment (vendethiel) [reply src/lexer.coffee]: Ok. LGTM then.

**Pull request merged**